cogs/tmpc.py

```diff
@@ -0,0 +1,276 @@
+        if not ctx.invoked_subcommand:
+            raise BadArgument
+        member: Union[Member, User] = ctx.author
+        logger.info(f'User="{member.name}#{member.discriminator}({member.id})", Command="{ctx.message.content}"')
+
+    @tmpc.command(pass_context=True)
+    async def keep(self, ctx: Context):
+        """
+        Makes a Study Channel stay for a longer time.
+        """
+        document = await self.check_tmpc_channel(ctx)
+        if type(document) == GamingChannel:
+            raise MayNotUseCommandError
+        if document.deleteAt:
+            document.deleteAt = None
+            embed = Embed(title="Turned off keep",
+                          description=f"This channel will no longer stay after everyone leaves")
+        else:
+            key = ConfigurationNameEnum.DEFAULT_KEEP_TIME.value
+            time_difference: tuple[int, int] = (await self.config_db.find_one({key: {"$exists": True}}))[key]
+            document.deleteAt = datetime.now() + timedelta(hours=time_difference[0], minutes=time_difference[1])
+            embed = Embed(title="Turned on keep",
+                          description=f"This channel will stay after everyone leaves for "
+                                      f"{time_difference[0]} hours and {time_difference[1]} minutes")
+
+        await self.study_db.update_one({DBKeyWrapperEnum.CHAT.value: document.channel_id}, document.document)
+        await ctx.reply(embed=embed)
+        await TmpChannelUtil.check_delete_channel(document.voice, self.study_db, logger)
+
+    @tmpc.command(pass_context=True)
+    async def lock(self, ctx: Context):
+        """
+        Locks a Study or Gaming Channel.
+        """
+        document = await self.check_tmpc_channel(ctx)
+        guild: Guild = ctx.guild
+
+        key = ConfigurationNameEnum.STUDENTY.value
+        verified = guild.get_role(
+            (await PrimitiveMongoData(CollectionEnum.ROLES).find_one({key: {"$exists": True}}))[key])
+
+        await document.voice.set_permissions(verified, view_channel=False, connect=False)
+
+        embed = Embed(title="Locked",
+                      description=f"Locked this channel. Now only the students that you can see on the "
+                                  f"right side can join and see the vc.\n"
+                                  f"If you are not a moderator, moderators can join and see this channel.")
+        await ctx.reply(embed=embed)
+
+    @tmpc.command(pass_context=True)
+    async def unlock(self, ctx: Context):
+        """
+        Unlocks a Study or Gaming Channel.
+        """
+        document = await self.check_tmpc_channel(ctx)
+        guild: Guild = ctx.guild
+
+        key = ConfigurationNameEnum.STUDENTY.value
+        verified = guild.get_role(
+            (await PrimitiveMongoData(CollectionEnum.ROLES).find_one({key: {"$exists": True}}))[key])
+
+        await document.voice.set_permissions(verified, view_channel=True, connect=True)
```

Comment:
view_channel should be handled with its own command.

Reply:
fixed